scrypto/src/modules/metadata.rs

```diff
@@ -0,0 +1,126 @@
+use crate::engine::scrypto_env::ScryptoEnv;
+use crate::runtime::*;
+use crate::*;
+use radix_engine_interface::api::node_modules::metadata::*;
+use radix_engine_interface::api::types::{NodeModuleId, ObjectId, RENodeId};
+use radix_engine_interface::api::{ClientObjectApi, ClientPackageApi};
+use radix_engine_interface::constants::METADATA_PACKAGE;
+use radix_engine_interface::data::scrypto::{scrypto_decode, scrypto_encode, ScryptoValue};
+use sbor::rust::prelude::ToOwned;
+use sbor::rust::string::String;
+use sbor::rust::vec::Vec;
+
+#[derive(PartialEq, Eq, Hash, Clone)]
+pub struct Metadata(pub ObjectId);
+
+impl Metadata {
+    pub fn new() -> Self {
+        let rtn = ScryptoEnv
+            .call_function(
+                METADATA_PACKAGE,
+                METADATA_BLUEPRINT,
+                METADATA_CREATE_IDENT,
+                scrypto_encode(&MetadataCreateInput {}).unwrap(),
+            )
+            .unwrap();
+        let metadata: Own = scrypto_decode(&rtn).unwrap();
+        Self(metadata.id())
+    }
+}
+
+impl MetadataObject for Metadata {
+    fn self_id(&self) -> (RENodeId, NodeModuleId) {
+        (RENodeId::Object(self.0), NodeModuleId::SELF)
+    }
+}
```

Comment:
Yup, its exists as a moveable node until it is attached to another node during globalization. Was planning on implementing the trait stuff for `AccessRules` in a future PR since that one would be a bigger refactor, but can add it to this PR as well.

Reply:
No problem, whichever is easier.